UserExperience_Remove_CssJs_Extension.php

```diff
@@ -195,32 +195,34 @@ public function w3tc_config_key_descriptor( $descriptor, $key ) {
 	/**
 	 * Performs actions on save.
 	 *
-	 * @since 2.7.0
+	 * @since X.X.X
 	 *
-	 * @param array $data Array of save data.
+	 * @static
 	 *
-	 * @return array
+	 * @param array $new_config New Config.
+	 * @param array $old_config Old Config.
 	 */
-	public function w3tc_save_options( $data ) {
-		$new_config = $data['new_config'];
-		$old_config = $data['old_config'];
-
-		$old_cssjs_includes = $old_config->get_array( array( 'user-experience-remove-cssjs', 'includes' ) );
-		$old_cssjs_singles  = $old_config->get_array( 'user-experience-remove-cssjs-singles' );
-		$new_cssjs_includes = $new_config->get_array( array( 'user-experience-remove-cssjs', 'includes' ) );
-		$new_cssjs_singles  = $new_config->get_array( 'user-experience-remove-cssjs-singles' );
+	public static function w3tc_config_ui_save_w3tc_userexperience( $new_config, $old_config ) {
+		$raw_cssjs_singles = Util_Request::get_array( 'user-experience-remove-cssjs-singles' );
 
-		foreach ( $new_cssjs_singles as $url => $pages ) {
-			if ( is_string( $pages['includes'] ) ) {
+		$new_cssjs_singles = array();
+		foreach ( $raw_cssjs_singles as $url => $pages ) {
+			if ( esc_url( $url ) && is_string( $pages['includes'] ) && ! empty( $pages['action'] ) ) {
```

Comment:
It looks like the `esc_url()` call is in the wrong place.  It's not escaping the data before it's used as an array key below.  I know that it was previously implemented in the settings that way, but it's not a good idea to use a URL as an array key.

Reply:
I pulled the esc_url and applied it to a variable so it's escaped in use. Using that value as a key was left as is, as changing it wouldn't be compatible with existing users and would require a restructure of all relevant code for the change in data structure storage